ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/wizards/exports/PluginExportWizardPage.java

```diff
@@ -71,6 +72,16 @@ private boolean hasBuildProperties(IPluginModelBase model) {
 		return file.exists();
 	}
 
+	private boolean hasPdeBndDescriptor(IPluginModelBase model) {
+		File bin = new File(model.getInstallLocation());
+		if (bin.exists()) {
+			File file = new File(bin.getParent(), ICoreConstants.BND_FILENAME_DESCRIPTOR);
+			return file.exists();
+		}
+		return false;
```

Comment:
Wouldn't it be better to operate on `IResources` here?
I.e. check if `model.getUnderlyingResource()` returns non null and then check for the `pde.bnd` file of that resource's project?

Reply:
Furthermore it is actually not necessary to check for the existance of the directory and file.
`File`, just like `Path` or `IFile/IFolder/...` are just handles/pointers and one can create them regardless of if they exist.
Just checking if the finally constructed `(I)File` is an existing file saves one call to the file-system (given the file exists).

Reply:
If you have a project handle one can also use `org.eclipse.pde.internal.core.natures.BndProject.isBndProject(IProject)` (that will only validate the nature) or `org.eclipse.pde.internal.core.bnd.BndProjectManager.getBndProject(IProject)` if also want to validate is has a *valid* instruction file.

Reply:
> Wouldn't it be better to operate on IResources here?
I.e. check if model.getUnderlyingResource() returns non null and then check for the pde.bnd file of that resource's project?

As far as I can see on the callerside the project is already known!

Reply:
> Furthermore it is actually not necessary to check for the existance of the directory and file. `File`, just like `Path` or `IFile/IFolder/...` are just handles/pointers and one can create them regardless of if they exist. Just checking if the finally constructed `(I)File` is an existing file saves one call to the file-system (given the file exists).

I have implemented the changes suggested by @laeubi , which has made these changes unnecessary, and also doesn't attempt file access.

Reply:
> If you have a project handle one can also use `org.eclipse.pde.internal.core.natures.BndProject.isBndProject(IProject)` (that will only validate the nature) or `org.eclipse.pde.internal.core.bnd.BndProjectManager.getBndProject(IProject)` if also want to validate is has a _valid_ instruction file.

I have implemented the changes, kindly look through and let me know @laeubi .